include/unifex/any_scheduler.hpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/any_unique.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/scheduler_concepts.hpp>
+#include <unifex/any_unique.hpp>
+#include <unifex/any_sender_of.hpp>
+#include <unifex/std_concepts.hpp>
+#include <unifex/type_index.hpp>
+#include <unifex/tag_invoke.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _any_sched {
+
+struct any_scheduler;
+
+template <typename Ret>
+struct copy_as_fn {
+  using type_erased_signature_t = Ret(const this_&);
+
+  template <typename T>
+  Ret operator()(const T& t) const {
+    if constexpr (tag_invocable<copy_as_fn, const T&>) {
+      return tag_invoke(*this, t);
+    } else {
+      return Ret{t};
+    }
+  }
+};
+template <typename Ret>
+inline constexpr copy_as_fn<Ret> copy_as{};
+
+inline constexpr struct get_type_id_fn {
+  using type_erased_signature_t = type_index(const this_&) noexcept;
+
+  template <typename T>
+  type_index operator()(const T& x) const noexcept {
+    if constexpr (tag_invocable<get_type_id_fn, const T&>) {
+      return tag_invoke(*this, x);
+    } else {
+      return type_id<T>();
+    }
+  }
+} get_type_id{};
```

Comment:
Naming of this should be `get_typeid` to match the keyword?
Or alternatively `get_type_index`?

Reply:
OK.